### PR TITLE
bpo-6634: [doc] clarify that sys.exit() does not always exit the interpreter

### PR DESCRIPTION
**Pull request by Vidhyavinu**
Updated the document at Exit() for main thread and non-primary threads.

https://bugs.python.org/issue6634

### CODE REVIEW

File: Doc/library/sys.rst [hunk 1]
```diff
@@ -449,10 +449,12 @@ always available.
 
 .. function:: exit([arg])
 
-   Exit from Python.  This is implemented by raising the :exc:`SystemExit`
+   Exit from Python.  If called in the main thread, this raises the :exc:`SystemExit`
```
Comment (iritkatriel) [inline Doc/library/sys.rst]: I think this always raises SystemExit, just the impact of that changes. (Check the code).

‘Exit from Python.’ Can be removed (it doesn’t always).

File: Doc/library/sys.rst [hunk 2]
```diff
@@ -449,10 +449,12 @@ always available.
...
    exception, so cleanup actions specified by finally clauses of :keyword:`try`
    statements are honored, and it is possible to intercept the exit attempt at
-   an outer level.
+   an outer level. When called from a thread other than the main thread, this
+   causes the thread to exit silently, nothing is printed and is equivalent to calling
+   :func:`thread.exit`.
```
Comment (iritkatriel) [inline Doc/library/sys.rst]: Have you verified this (through testing or from the code)?

Comment (Vidhyavinu) [reply Doc/library/sys.rst]: I tried with a sample code. Exception is raised from both main and non-primary thread. The doc is updated. Please review.

File: Doc/library/sys.rst [hunk 3]
```diff
@@ -449,10 +449,9 @@ always available.
 
 .. function:: exit([arg])
 
-   Exit from Python.  This is implemented by raising the :exc:`SystemExit`
-   exception, so cleanup actions specified by finally clauses of :keyword:`try`
-   statements are honored, and it is possible to intercept the exit attempt at
-   an outer level.
+   This is implemented by raising the :exc:`SystemExit` exception, so cleanup 
+   actions specified by finally clauses of :keyword:`try` statements are honored, 
+   and it is possible to intercept the exit attempt at an outer level.
```
Comment (iritkatriel) [inline Doc/library/sys.rst]:
```suggestion
   Raise a :exc:`SystemExit` exception, signaling an intention to exit the interpreter.
   Cleanup actions specified by finally clauses of :keyword:`try` statements are honored, 
   and it is possible to intercept the exit attempt at an outer level. 
```

File: Doc/library/sys.rst (hunk 2 above)
Comment (Vidhyavinu) [reply Doc/library/sys.rst]: The edit looks good. Thanks.

File: Doc/library/sys.rst [hunk 4]
```diff
@@ -449,10 +449,9 @@ always available.
 
 .. function:: exit([arg])
 
-   Exit from Python.  This is implemented by raising the :exc:`SystemExit`
-   exception, so cleanup actions specified by finally clauses of :keyword:`try`
-   statements are honored, and it is possible to intercept the exit attempt at
-   an outer level.
+   Raise a :exc:`SystemExit` exception, signaling an intention to exit the interpreter.
+   Cleanup actions specified by finally clauses of :keyword:`try` statements are honored, 
+   and it is possible to intercept the exit attempt at an outer level.
```
Comment (iritkatriel) [inline Doc/library/sys.rst]: I'd move the second sentence ("Cleanup...") down - just after the sentence that ends in line 471.

Comment (Vidhyavinu) [reply Doc/library/sys.rst]: Yup. The document is updated. please review.

File: Doc/library/sys.rst [hunk 5]
```diff
@@ -471,6 +468,9 @@ always available.
    the process when called from the main thread, and the exception is not
    intercepted.
 
+   Cleanup actions specified by finally clauses of :keyword:`try` statements are honored, 
+   and it is possible to intercept the exit attempt at an outer level.
```
Comment (iritkatriel) [inline Doc/library/sys.rst]:
```suggestion
   intercepted.  Cleanup actions specified by finally clauses of :keyword:`try` statements
   are honored, and it is possible to intercept the exit attempt at an outer level. 
```

**Review by iritkatriel — APPROVED** [review]

File: Doc/library/sys.rst [hunk 6]
```diff
@@ -469,8 +466,9 @@ always available.
 
    Since :func:`exit` ultimately "only" raises an exception, it will only exit
    the process when called from the main thread, and the exception is not
-   intercepted.
-
+   intercepted.  Cleanup actions specified by finally clauses of :keyword:`try` statements
+   are honored, and it is possible to intercept the exit attempt at an outer level. 
+
```
Comment (iritkatriel) [inline Doc/library/sys.rst]:
```suggestion
   are honored, and it is possible to intercept the exit attempt at an outer level.

```

File: Doc/library/sys.rst [hunk 7]
```diff
@@ -469,7 +466,8 @@ always available.
 
    Since :func:`exit` ultimately "only" raises an exception, it will only exit
    the process when called from the main thread, and the exception is not
-   intercepted.
+   intercepted.  Cleanup actions specified by finally clauses of :keyword:`try` statements
```
Comment (iritkatriel) [inline Doc/library/sys.rst]:
```suggestion
   intercepted. Cleanup actions specified by finally clauses of :keyword:`try` statements
```

**Pull request merged**